Monokai Pane

Xyn0gen opened the pull request:
Based off the display in Flyingtuna and a bit of ryuk. Uses the monokai colour palette.
> *Song Selection*\
<img src="https://i.imgur.com/1fb4BEe.png" width="500">\
>*Gameplay 1*\
<img src="https://i.imgur.com/WIzTo99.png" width="500">\
>*Gameplay 2*\
><img src="https://i.imgur.com/OyoMzE3.png" width="500">

Xyn0gen committed:
Add files via upload

it is literally a copy of flyingtuna's song display with a pp and hit counter that is also literally the same design

Xyn0gen committed:
changed title

Xyn0gen committed:
now shows current rank every 10 seconds

Xyn0gen committed:
added new ranked status for difficulty display

Xyn0gen committed:
cleaned up some spaghetti

Xyn0gen committed:
fixed qualified colour

Xyn0gen committed:
tunadisplay but fancier?

Xyn0gen committed:
converted everything to rem

Xyn0gen committed:
fixed stuff in tunadisplayv2 and made monokaipane

Xyn0gen committed:
Update README.md

Xyn0gen committed:
Update README.md

Xyn0gen committed:
Update README.md

Xyn0gen committed:
i actually dont understand how to make a readme

Xyn0gen committed:
fixed text shadows clipping and others

Xyn0gen committed:
removed tunadisplay

Xyn0gen committed:
changed img, removed unused vars

Xyn0gen committed:
unused font removed

l3lackShark commented:
Thanks for your contribution! Our Front-Ender will review your code within the next couple of days.

l3lackShark commented:
Consider reducing the width, as it will probably be too wide for most overlays.

Xyn0gen committed:
default width is now 32rem

Xyn0gen commented:
![image](https://user-images.githubusercontent.com/6222794/93663690-0b0e6380-faad-11ea-8431-ab81f6bf3e49.png)
It is now 512px

Xyn0gen commented:
Also I don't know if it's a good thing or not but I used rem for all the sizes. I thought that it would make things more aligned if you change stuff.

Dartandr commented on MonokaiPane/index.html:
```diff
@@ -0,0 +1,62 @@
+          </div>
+        </div>
+      </div>
+      <img src="img.png" id="bg">
```
Why do you use some placeholder img you can just leave src empty

Dartandr commented on MonokaiPane/index.html:
```diff
@@ -0,0 +1,62 @@
+  <link rel="stylesheet" href="css/style.css">
+  <link rel="stylesheet" href="css/odometr.css">
+  <title>MonokaiPane</title>
+  <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.5.1/jquery.min.js"></script>
+  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.14.0/css/all.min.css"/>
```
I'm not recommend to use links to external sources for resources files because some people may want to try you counter without internet or so

Dartandr commented on MonokaiPane/css/style.css:
```diff
@@ -0,0 +1,207 @@
+@import url('https://fonts.googleapis.com/css2?family=Ubuntu:ital,wght@0,300;0,400;0,500;0,700;1,300;1,400;1,500;1,700&display=swap');
```
Same in as in index.html

Dartandr commented:
> Also I don't know if it's a good thing or not but I used rem for all the sizes. I thought that it would make things more aligned if you change stuff.

it's totally fine

Xyn0gen committed:
localised jquery, fa, ubuntu

Xyn0gen committed:
removed placeholder

Xyn0gen committed:
removed placeholder

Xyn0gen commented:
that took longer than expected

Dartandr reviewed (approved):
lgtm

Xyn0gen commented:
actually seems like the timer for switching between showing the rank and song name/artist is bugging out. trying to see whats going on

Xyn0gen committed:
fixed bug where rank display pops out of timer

so state 14 is multi end screen and that causes the interval to not get cleared

Xyn0gen committed:
added progress bar and cleaned up unused css

Xyn0gen committed:
no idea how the hdfl check worked before this change

Xyn0gen committed:
forgot to delete placeholder locally

Xyn0gen committed:
forgot to delete this testing local fonts

Xyn0gen committed:
removed unused fonts and rank placeholder

l3lackShark commented:
Consider handling long beatmap names
![image](https://user-images.githubusercontent.com/20248967/93850970-adeef980-fcc8-11ea-944c-d5789352f77c.png)

Xyn0gen commented:
I'll work on adding a marquee effect when the text exceeds the bounding area.

osugenics commented:
looks very nice :)

Xyn0gen committed:
added scrolling effect for long titles and artists

Xyn0gen commented:
Suggestions for a better way to do it is welcomed but this is what I've come up with.

Xyn0gen committed:
changed the order of code in index.js

Xyn0gen commented:
I think setting the title before adding the marquee class makes it less jumpy.

Xyn0gen committed:
added speed calculator to control speed of marquee

Xyn0gen committed:
Delete jquery.marquee.min.js

l3lackShark commented:
Please purge your grade logic and replace with our JSON token. 
![image](https://user-images.githubusercontent.com/20248967/94269355-fcfe8e00-ff57-11ea-8575-3e2aaea695f2.png)

Night build that has this token could be obtained from here: https://github.com/l3lackShark/gosumemory/runs/1165884656

Xyn0gen committed:
removed rank calculator and dependancy on jquery

l3lackShark commented:
Is this ready for merge?

Xyn0gen commented:
Yeah

l3lackShark committed:
Update sizes

l3lackShark commented:
Thanks again
